app/src/main/java/com/asfoundation/wallet/repository/SharedPreferencesRepository.kt

```diff
@@ -137,4 +138,21 @@ class SharedPreferencesRepository(private val pref: SharedPreferences) : Prefere
   }
 
   override fun getNumberOfTimesOnHome(): Int = pref.getInt(NUMBER_OF_TIMES_IN_HOME, 0)
+
+  override fun hasBeenInPromotionsScreen(): Boolean =
+      pref.getBoolean(HAS_BEEN_IN_PROMOTIONS_SCREEN, false)
+
+  override fun setHasBeenInPromotionsScreen() {
+    pref.edit()
+        .putBoolean(HAS_BEEN_IN_PROMOTIONS_SCREEN, true)
+        .apply()
+  }
+
+  override fun hasSeenVoucherTooltip(): Boolean = pref.getBoolean(HAS_SEEN_VOUCHER_TOOLTIP, false)
+
+  override fun setHasSeenVoucherTooltip() {
+    pref.edit()
+        .putBoolean(HAS_SEEN_VOUCHER_TOOLTIP, true)
+        .apply()
+  }
```

Comment:
We are starting to have too many of these "hasSeen" "hasBeen" methods. Maybe we should create a dedicated SharedPrefRepo for these. ImpressionsPreferenceRepository.kt. Feel free to come up with a better name.